src/main/java/com/thealgorithms/datastructures/graphs/WelshPowell.java

```diff
@@ -0,0 +1,101 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+
+/*
+ *  The Welsh-Powell algorithm is a graph coloring algorithm
+ *  used for coloring a graph with the minimum number of colors.
+ *  https://en.wikipedia.org/wiki/Graph_coloring
+ */
+
+public class WelshPowell {
+
+    public static class WPGraph {
+        private int numVer; // Number of vertices in the graph
+        private HashSet<Integer>[] adjLists; // Adjacency lists for the graph using HashSet
+
+        public WPGraph(int vertices) {
+            numVer = vertices;
+            adjLists = new HashSet[vertices];
+            for (int i = 0; i < vertices; i++) {
+                adjLists[i] = new HashSet<>();
+            }
+        }
+
+        public void addEdge(int nodeA, int nodeB) {
+            adjLists[nodeA].add(nodeB);
```

Comment:
I would suggest at lest that:
```suggestion
            assert (0 <= nodeA && nodeA <= numVer);
            assert (0 <= nodeB && nodeB <= numVer);
            adjLists[nodeA].add(nodeB);
```